lib/elixir/lib/uri.ex

```diff
@@ -977,6 +977,60 @@ defmodule URI do
       %{uri | query: uri.query <> "&" <> query}
     end
   end
+
+  @doc """
+  Appends `path` to the given `uri`.
+
+  Path must start with "/" and cannot contain additional url components like
+  fragments or query strings.
+
+  ## Examples
+
+      iex> URI.append_path(URI.parse("http://example.com/foo/?x=1"), "/my-path") |> URI.to_string()
+      "http://example.com/foo/my-path?x=1"
+
+      iex> URI.append_path(URI.parse("http://example.com"), "my-path")
+      ** (ArgumentError) path must start with "/", got: "my-path"
+  """
+  @doc since: "1.15.0"
+  @spec append_path(t(), binary()) :: t()
```

Comment:
Nitpick, seems like we tend to use `String.t()` more in this module:

```suggestion
  @spec append_path(t(), String.t()) :: t()
```